### PR TITLE
EOS-28475 Hare need to stop using 'cortx>motr>interface_type' key

### PR DESCRIPTION
**Pull request by SwapnilGaonkar7**
Problem: Protocol information needs to be retrieved from endpoint flag

Solution:
Modified code to read 'cortx>hare>hax>endpoints' which is in the format
'<protocol>://<hostname>:<port>' and getting interface type(protocol) from it

Signed-off-by: Swapnil Gaonkar <swapnil.gaonkar@seagate.com>

Tested 'config' stage by adding following keys in confstore
```
cortx:
  hare:
    hax:
      endpoints:
      - tcp://cortx-data-headless-svc-ssc-vm-g3-rhev4-2623:22001
      - tcp://cortx-data-headless-svc-ssc-vm-g2-rhev4-2784:22001
      - tcp://cortx-data-headless-svc-ssc-vm-g2-rhev4-2785:22001
      - tcp://cortx-server-headless-svc-ssc-vm-g2-rhev4-2784:22001
      - tcp://cortx-server-headless-svc-ssc-vm-g3-rhev4-2623:22001
      - tcp://cortx-server-headless-svc-ssc-vm-g2-rhev4-2785:22001
      - https://cortx-data-headless-svc-ssc-vm-g3-rhev4-2623:22001
```

### CODE REVIEW

File: provisioning/miniprov/hare_mp/cdf.py [hunk 1]
```diff
@@ -312,11 +312,31 @@ def _get_iface(self, machine_id: str) -> str:
...
+        endpoints = self.provider.get(
+            'cortx>hare>hax>endpoints',
             allow_null=True)
+
+        if endpoints is None:
+            return None
+
+        hostname = self.utils.get_hostname(machine_id)
+
+        iface = None
```
Comment (mssawant) [inline provisioning/miniprov/hare_mp/cdf.py:325]: let's use `proto` instead of `iface`? `proto` seems more relevant.

**Review by suvratjoshi — APPROVED** [review]

File: provisioning/miniprov/hare_mp/cdf.py [hunk 2]
```diff
@@ -312,11 +312,31 @@ def _get_iface(self, machine_id: str) -> str:
...
+        # e.g. endpoints:
+        #      - tcp://data1-node1:22001  # For motr and Hax communication
+        #      - tcp://data1-node2:22001  # For motr and Hax communication
+        for e in endpoints:
+            key = e.split(':')
+
+            if key[0] == 'https':
+                continue
+
+            if key[1].split('/')[2] == hostname:
```
Comment (Shreya-18) [inline provisioning/miniprov/hare_mp/cdf.py:336]: Just a suggestion: can we check directly ?
```suggestion
            if hostname in key[1]:
	         iface = key[0]
```

Comment (SwapnilGaonkar7) [reply provisioning/miniprov/hare_mp/cdf.py:336]: `if key[1].split('/')[2] == hostname:`
@Shreya-18 Thanks for the input. Even though original solution looks complex as compare to solution suggested it checks for exact format.
@mssawant Any suggestion here?

File: provisioning/miniprov/hare_mp/cdf.py (hunk 1 above)
Comment (SwapnilGaonkar7) [reply provisioning/miniprov/hare_mp/cdf.py]: Done

**Review by vaibhavparatwar — APPROVED** [review]

**Review by Shreya-18 — APPROVED** [review]

File: provisioning/miniprov/hare_mp/cdf.py (hunk 2 above)
Comment (mssawant) [reply provisioning/miniprov/hare_mp/cdf.py:339]: Looks fine.

**Pull request merged**